Composite/Search/DocumentSources/CmsPageDocumentSource.cs

```diff
@@ -16,14 +16,18 @@
 
 namespace Composite.Search.DocumentSources
 {
-    internal class CmsPageDocumentSource : ISearchDocumentSource
+    public class CmsPageDocumentSource : ISearchDocumentSource
     {
         const string LogTitle = nameof(CmsPageDocumentSource);
 
-        private readonly List<IDocumentSourceListener> _listeners = new List<IDocumentSourceListener>();
+        protected List<IDocumentSourceListener> _listeners = new List<IDocumentSourceListener>();

```

Comment:
Bad practice to make fields accessible from outside the class !!

Should be a property.